### ISSUE TITLE
Sync Stops Mid-Run

### ISSUE THREAD

**Issue by zluckytraveler** [Taxel/PlexTraktSync#805]
**PlexTraktSync Version: 0.18.20
Python Version: 3.10.2 (main, Feb  2 2022, 08:42:42) [Clang 13.0.0 (clang-1300.0.29.3)]
Plex API Version: 4.9.1
Cache Dir: /Users/Lucky/Library/Caches/PlexTraktSync
Config Dir: /Users/Lucky/Library/Application Support/PlexTraktSync
Log Dir: /Users/Lucky/Library/Logs/PlexTraktSync
Plex Server version: 1.25.5.5492-12f6b8c83, updated at: 2022-02-11 05:51:22**

I just installed plextraktsync via pipx, and the sync will stop in the middle of its run. I have included the output from terminal, that I get once the run stops, and also a debug log.

```Traceback (most recent call last):
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/urllib3/connectionpool.py", line 449, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/urllib3/connectionpool.py", line 444, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1374, in getresponse
    response.begin()
  File "/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 318, in begin
    version, status, reason = self._read_status()
  File "/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 279, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/socket.py", line 705, in readinto
    return self._sock.recv_into(b)
TimeoutError: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/requests/adapters.py", line 440, in send
    resp = conn.urlopen(
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/urllib3/connectionpool.py", line 785, in urlopen
    retries = retries.increment(
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/urllib3/util/retry.py", line 550, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/urllib3/packages/six.py", line 770, in reraise
    raise value
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/urllib3/connectionpool.py", line 451, in _make_request
    self._raise_timeout(err=e, url=url, timeout_value=read_timeout)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/urllib3/connectionpool.py", line 340, in _raise_timeout
    raise ReadTimeoutError(
urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host=): Read timed out. (read timeout=30)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/Lucky/.local/bin/plextraktsync", line 8, in <module>
    sys.exit(cli())
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plextraktsync/cli.py", line 28, in wrap
    cmd(*args, **kwargs)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plextraktsync/commands/sync.py", line 58, in sync
    runner.sync(walker=w, dry_run=config.dry_run)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plextraktsync/sync.py", line 71, in sync
    for episode in walker.find_episodes():
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plextraktsync/walker.py", line 247, in find_episodes
    yield from self.episode_from_show(show)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plextraktsync/walker.py", line 282, in episode_from_show
    me = self.mf.resolve_any(pe, show)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plextraktsync/media.py", line 160, in resolve_any
    m = self.resolve_guid(guid, show)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plextraktsync/media.py", line 168, in resolve_guid
    logger.warning(f"{guid.pm.item}: Skipping guid {guid} because provider {guid.provider} has no external Id")
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plexapi/video.py", line 814, in __repr__
    '%s-%s' % (self.grandparentTitle.replace(' ', '-')[:20], self.seasonEpisode),
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plexapi/base.py", line 457, in __getattribute__
    value = super(PlexPartialObject, self).__getattribute__(attr)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plexapi/video.py", line 851, in seasonEpisode
    return 's%se%s' % (str(self.seasonNumber).zfill(2), str(self.episodeNumber).zfill(2))
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plexapi/base.py", line 457, in __getattribute__
    value = super(PlexPartialObject, self).__getattribute__(attr)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plexapi/video.py", line 839, in episodeNumber
    return self.index
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plexapi/base.py", line 471, in __getattribute__
    self._reload(_autoReload=True)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plexapi/base.py", line 350, in _reload
    data = self._server.query(key)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/plexapi/server.py", line 672, in query
    response = method(url, headers=headers, timeout=timeout, **kwargs)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/requests/sessions.py", line 542, in get
    return self.request('GET', url, **kwargs)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/requests_cache/session.py", line 115, in request
    return super().request(method, url, headers=headers, **kwargs)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/requests/sessions.py", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/requests_cache/session.py", line 147, in send
    response = self._send_and_cache(request, actions, **kwargs)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/requests_cache/session.py", line 189, in _send_and_cache
    response = super().send(request, **kwargs)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/requests/sessions.py", line 645, in send
    r = adapter.send(request, **kwargs)
  File "/Users/Lucky/.local/pipx/venvs/plextraktsync/lib/python3.10/site-packages/requests/adapters.py", line 532, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPConnectionPool(host=): Read timed out. (read timeout=30)

**Comment by glensc** [comment]
@zluckytraveler you need to revoke your trakt api keys because you have shared them here publicly here along with the logfile

**Comment by zluckytraveler** [comment]
@glensc this is the complete output, I copy pasted directly from terminal.

**Comment by glensc** [comment]
perhaps the terminal has scroll which you can see, but plextraktsytnc starts with printing it's version, and you do not have it, so it can't be complete output.

anyway, your plex server is giving a timeout error, you could try to increase plex timeout:

see previous responses to the same timeout errors:
- https://github.com/Taxel/PlexTraktSync/issues/742#issuecomment-1019561555
- https://github.com/Taxel/PlexTraktSync/issues/742#issuecomment-1019561796
- https://github.com/Taxel/PlexTraktSync/issues/742#issuecomment-1020028169

**Comment by zluckytraveler** [comment]
Here is a complete terminal output of the entire sync. I'll see if i'm able to change the timeout, although I'm very novice and don't quite understand how to use plexapi and get it working, also not to mention I'm on a appbox, so not sure if its even possible to change those settings.

[Terminal Output.zip](https://github.com/Taxel/PlexTraktSync/files/8056865/Terminal.Output.zip)
.

**Comment by glensc** [comment]
If plex is giving a timeout for the same item, it's time to look at plex own logs. you can try `inspect` subcommand if you're able to figure out id, maybe resolving it once for that item will plex cache results internally. i don't know how plex operates. perhaps plex forums is better place to ask (don't forget to cross link if you posted there).

I don't know what is appbox, but according to docs you need to create and edit `~/.config/plexapi/config.ini`:
- https://python-plexapi.readthedocs.io/en/latest/configuration.html

**Comment by glensc** [comment]
I looked at the trace and the exception happened on handling another error, where i tried to print that an item failed:
```
    logger.warning(f"{guid.pm.item}: Skipping guid {guid} because provider {guid.provider} has no external Id")
```

you can try this branch as a mitigation for your problem:
- https://github.com/Taxel/PlexTraktSync/pull/806

**Comment by zluckytraveler** [comment]
Unfortunately no files were created in the installation and nothing can be found regarding it, and their support is quite dismal, so it'd seem that plexapi is out of question for adjusting the timeout.

I'm fairly new to using github still. What do you mean by trying out that branch, and how would I use it?

**Comment by glensc** [comment]
I don't understand in what way plex support and python plex api are related.

**Comment by zluckytraveler** [comment]
This is in reference to plexapi support, not plex.

**Comment by glensc** [comment]
what plexapi support you want? you are unable to create the file or set the environment variable?

did you read the document? it's in plain English:
- https://python-plexapi.readthedocs.io/en/latest/configuration.html

> The default configuration file path is `~/.config/plexapi/config.ini` which can be overridden by setting the environment variable `PLEXAPI_CONFIG_PATH` with the file path you desire. 

and

> All configuration values can be set or overridden via environment variables. The environment variable names are in all upper case and follow the format `PLEXAPI_<SECTION>_<NAME>`.
